src/seedu/addressbook/AddressBook.java

```diff
@@ -329,7 +364,13 @@ private static boolean isValidFilePath(String filePath) {
      */
     private static boolean hasValidParentDirectory(Path filePath) {
         Path parentDirectory = filePath.getParent();
-        return parentDirectory == null || Files.isDirectory(parentDirectory);
+        boolean hasParentDirectory;
+        if(parentDirectory == null){
+        	hasParentDirectory = false;
+        }else{
+        	hasParentDirectory = true;
+        }
```

Comment:
The if statement violate our coding standard, you miss some spaces. Please refer to our coding standard.